src/main/java/dev/morling/onebrc/CalculateAverage_isolgpus.java

```diff
@@ -123,38 +117,49 @@ private static MeasurementCollector[] handleChunk(FileChannel channel, long esti
 
         try {
 
-            while (i <= lengthOfChunk || !parsingName) {
+            while (i <= lengthOfChunk) {
+                int nameSum = 0;
+                int hashResult = 0;
+                int nameStart;
                 byte aChar;
-                if (parsingName) {
-
-                    while ((aChar = r.get()) != SEPERATOR) {
-                        nameBuffer[nameBufferIndex++] = aChar;
-                        nameSum += aChar;
-                        hashResult = 31 * hashResult + aChar;
+                nameStart = i;
+                int nameBufferIndex = 0;
+                int valueIndex = 0;
+
+                // optimistically assume that the name is at least 4 bytes
+                int firstInt = r.getInt();
```

Comment:
Is this just slow when that assumption breaks, or will it fail?

Reply:
Just slow, on line 143 if it notices it has gone past the separator without noticing it will rewind back to the beginning of the line and read it byte for byte.

Reply:
Ah yes, I see. Thx!